Allow usage with nix daemon

AleksanderGondek opened the pull request:
This commit removes the restriction of running the project only via single-user nix installation.
To ensure disruptor is run with appropriate settings, a comparison is made betweeen global nix configuration and those specified in scripts/nix.conf file - any differences is clearly communicated to user

AleksanderGondek committed:
Allow usage with nix deamon

This commit removes the restriction of running the project only via single-user nix installation.
To ensure disruptor is run with appropriate settings, a comparison is made betweeen global nix configuration and those specified in scripts/nix.conf file - any differences is clearly communicated to user

AleksanderGondek commented:
Example of warning:
```sh
./nix-shell.sh
The configuration of host nix daemon differs from the expected one.
These are the offending settings:
{ keep-outputs = true; require-sigs = false; }
---
Please update your configuration otherwise the shell may not work as intended.
direnv: loading ~/projects/disruptor/.envrc

[nix-shell:~/projects/disruptor]$
```

AleksanderGondek committed:
[CI] MacOS nix installation: add settings

AleksanderGondek committed:
Human friendly output of nix config diff

r2r-dev commented:
![Alt Text](https://c.tenor.com/KnaDXgxBV2QAAAAC/takes-glasses-off-deal-with-it.gif)

r2r-dev commented on scripts/shell.sh:
```diff
@@ -97,6 +97,43 @@ preflightCheck() {
+  nix show-config --json > "${global_nix_conf_settings}" 2>/dev/null
+  # shellcheck disable=SC2181
+  if [ $? -ne 0 ]; then
+   fail "nix show-config command has failed. Please enable it by following this steps: https://nixos.wiki/wiki/Nix_command"
```
Why not add `--experimental-features nix-command` to the command above?

Also this condition can be reached in cases when installed nix version is below 2.4.

AleksanderGondek committed:
Improve nix show-config call

This change adds '--experimental-features nix-command' param to nix show-config invocation to address invocations of privileged users who have not yet enabled nix-command

r2r-dev commented:
I like where this is heading. I'm ok with merging these changes. However, I think large chunks of this solution could be moved to `shell.nix` entirely. Some references:
https://github.com/NixOS/nixpkgs/blob/420a0cce9358acf8b9c9c2ae3a0e421097f84c50/nixos/modules/services/misc/nix-daemon.nix#L72
https://github.com/cheronobyl/nixflk/blob/27513eeed119ccb82eee0abb60b93f0cd9758473/shell.nix#L25
